src/rgw/driver/posix/rgw_sal_posix.cc

```diff
@@ -624,52 +630,70 @@ int POSIXBucket::list(const DoutPrefixProvider* dpp, ListParams& params, int max
       return 0;
     }
 
-    std::string_view vname = name;
-    //if (vname.find(params.delim) == std::string_view::npos) {
-      //// Delimiter doesn't match; skip
-      //return 0;
-    //}
-
-    if (!vname.starts_with(params.prefix)) {
-      // Prefix doesn't match; skip
-      ldpp_dout(dpp, 0) << "C3PO vname: " << vname << " prefix: \"" << params.prefix << "\"" << dendl;
-      return 0;
-    }
-
-    if (max-- <= 0) {
-      return -EAGAIN;
-    }
-
     ret = statx(dir_fd, name, AT_SYMLINK_NOFOLLOW, STATX_ALL, &stx);
     if (ret < 0) {
       ret = errno;
       ldpp_dout(dpp, 0) << "ERROR: could not stat object " << name << ": "
 	<< cpp_strerror(ret) << dendl;
-      results.objs.clear();
       return -ret;
     }
 
     if (S_ISREG(stx.stx_mode) || S_ISDIR(stx.stx_mode)) {
-      rgw_bucket_dir_entry *e = new rgw_bucket_dir_entry;
-      e->key.name = decode_name(name);
-      e->ver.pool = 1;
-      e->ver.epoch = 1;
-      e->exists = true;
-      e->meta.category = RGWObjCategory::Main;
-      e->meta.size = stx.stx_size;
-      e->meta.mtime = ceph::real_clock::from_time_t(stx.stx_mtime.tv_sec);
-      e->meta.owner = std::to_string(stx.stx_uid); // TODO convert to owner name
-      e->meta.owner_display_name = std::to_string(stx.stx_uid); // TODO convert to owner name
-      e->meta.accounted_size = stx.stx_blksize * stx.stx_blocks; // TODO Won't work for mpobj
-      e->meta.storage_class = RGW_STORAGE_CLASS_STANDARD;
-      e->meta.appendable = true;
-
-      results.objs.push_back(*e);
+      rgw_bucket_dir_entry bde{};
+      bde.key.name = decode_name(name);
+      bde.ver.pool = 1;
+      bde.ver.epoch = 1;
+      bde.exists = true;
+      bde.meta.category = RGWObjCategory::Main;
+      bde.meta.size = stx.stx_size;
+      bde.meta.mtime = ceph::real_clock::from_time_t(stx.stx_mtime.tv_sec);
```

Comment:
if we have a full resolution timespec, why not use it?

Reply:
We could.  statx() doesn't use a normal timestamp, it has it's own struct, so I would have had to open-code it, and I was lazy.

Reply:
ah, ok